app/js/components/mixins/ModalEffects.react.js

```diff
@@ -1,23 +1,40 @@
 import React from 'react'
 
+/*
+  Here's the ultimate modal creator :)
```

Comment:
The setTimeout bit is optional, only when `autoShow` is set to true does the function run. I understand your concern about memory leaks though. I'm open to refractor suggestion ideas.